gobblin-utility/src/main/java/org/apache/gobblin/util/PullFileLoader.java

```diff
@@ -145,28 +146,36 @@ public Config loadPullFile(Path path, Config sysProps, boolean loadGlobalPropert
   }
 
   /**
-   * Find and load all pull files under a base {@link Path} recursively.
+   * Find and load all pull files under a base {@link Path} recursively in an order sorted by last modified date.
    * @param path base {@link Path} where pull files should be found recursively.
    * @param sysProps A {@link Config} used as fallback.
    * @param loadGlobalProperties if true, will also load at most one *.properties file per directory from the
    *          {@link #rootDirectory} to the pull file {@link Path} for each pull file.
    * @return The loaded {@link Config}s.
    */
-  public Collection<Config> loadPullFilesRecursively(Path path, Config sysProps, boolean loadGlobalProperties) {
+  public List<Config> loadPullFilesRecursively(Path path, Config sysProps, boolean loadGlobalProperties) {
     try {
       Config fallback = sysProps;
       if (loadGlobalProperties && PathUtils.isAncestor(this.rootDirectory, path.getParent())) {
         fallback = loadAncestorGlobalConfigs(path.getParent(), fallback);
       }
-      return loadPullFilesRecursivelyHelper(path, fallback, loadGlobalProperties);
+      return getSortedConfigs(loadPullFilesRecursivelyHelper(path, fallback, loadGlobalProperties));
```

Comment:
This is doing sorting at each recursion level. You can optimize it to single sort when the listing is complete?

Reply:
I think this is at outer level. If I would be sorting inside loadPullFilesRecursivelyHelper(), it would have been 'at each recursion level'. Right?

Reply:
loadPullFilesRecursively(getSortedConfigs(loadPullFilesRecursively(getSortedConfigs(loadPullFilesRecursively( .. )))))

Reply:
as discussed, we can ignore the above comment since getSortedConfigs is calling loadPullFilesRecursivelyHelper and not loadPullFilesRecursively